### ISSUE TITLE
Possible to macro or keybind a saved note?

### ISSUE THREAD

**Issue by Sahasrahla** [GurliGebis/WoWAddon-PasteNG#29]
I have a saved note that I frequently reuse. In my use-case, it is a list of TomTom coordinates, too long for a standard macro. It marks all the rare mobs in The Emerald Dream that drop druid form customizations. I fly to each TomTom beacon, it disappears, I circle back to the camp at the center. Log back in an hour or so later and repeat the process.

Is there a way I could activate the saved note with a keybind or macro on my bars rather than opening the PasteNG menu, Loading the saved note, and then clicking Paste and Close?

**Comment by GurliGebis** [comment]
Would a command like `/paste send "Name of saved paste"` be okay?

**Comment by Sahasrahla** [comment]
Yes, that seems perfect for me.

When saving a note, does PasteNG retain which chat channel to use for destination along with the text, or is that chosen at the time of loading the saved note and pasting? In my use-case, default chat channel works fine, but maybe an optional switch so that the saved note can be pasted to other chat channels via command would be good. I imagine something like a saved blurb for a profession crafter's advertisement, and they would want to quickly be able to paste the same advertisement to Trade chat as well as Services chat or something.

**Comment by GurliGebis** [comment]
I doesn't save the selected channel.
I can add an optional parameter, so you can define which channel to use

**Comment by GurliGebis** [comment]
I have just pushed `11.0.5-20241121-1` which implements this.

```
/pasteng send "Saved Paste name" - Send a save paste to the default channel
/pasteng send "Saved Paste name" channel - Send a save paste to a specific channel
```

**Comment by GurliGebis** [comment]
If you don't have the legacy Paste addon installed, you can use `/paste` instead 😊

**Comment by Sahasrahla** [comment]
This worked brilliantly for me. I was able to quickly load my TomTom waypoints from the command line, and can also confirm putting that same command into a macro worked as intended when clicked from my action bar. Thank you for your quick implementation on a very useful feature.

**Comment by GurliGebis** [comment]
You're welcome - was pretty easy to add.
If you have other ideas, just let me know 😊

**Comment by Sahasrahla** [comment]
I was too quick in my prior assessment and have found an issue. When I tested before, I now realize, I had opened the PasteNG UI to confirm the name of the note before I tested the command line and macro usage. I've now found that if I load into a character and immediately type `/paste send "Druid Form Rares"` into the console, I am prevented from pressing Enter to submit the entry. Similarly if I click my macro button it does nothing. The behavior is also recreated after performing a /reload.

Once I open the PasteNG UI with `/paste show`, then both the console and macro begin working as intended. I do not have to interact with the PasteNG UI, merely the act of opening the UI seems to enable the "send" command to work from console.
